### ISSUE TITLE
All tests require full timeout period to pass after twister overhaul when executed on HW platform

### ISSUE THREAD

**Issue by evgeniy-paltsev** [zephyrproject-rtos/zephyr#45904]
**Describe the bug**
All tests require full timeout period to pass after twister overhaul, even if actually test executes really fast (like hello-world sample).

Here is log example of running 3 tests:
```
INFO    - 3 test scenarios (3 configurations) selected, 0 configurations discarded due to filters.
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - 1/3 hsdk                      samples/hello_world/sample.basic.helloworld        PASSED (device 63.116s)
INFO    - 2/3 hsdk                      samples/philosophers/sample.kernel.philosopher.stacks PASSED (device 60.610s)
INFO    - 3/3 hsdk                      tests/kernel/common/kernel.common                  PASSED (device 63.128s)

INFO    - 3 of 3 test configurations passed (100.00%), 0 failed, 0 skipped with 0 warnings in 207.48 seconds
INFO    - In total 47 test cases were executed, 0 skipped on 1 out of total 456 platforms (0.22%)
INFO    - 3 test configurations executed on platforms, 0 test configurations were only built.
```

Twister options used:
```
./scripts/twister -v --ninja --jobs 4 --disable-warnings-as-errors --test samples/hello_world/sample.basic.helloworld --test samples/philosophers/sample.kernel.philosopher.stacks --test tests/kernel/common/kernel.common --platform hsdk --west-runner=mdb-hw --device-testing --device-serial /dev/ttyUSB0 --west-flash=--dig-device=JtagHs2 -W --pre-script /home/paltsev
/CodeData/zephyr/pre-script.sh
```

Full logs and twister reports:
[artefacts.zip](https://github.com/zephyrproject-rtos/zephyr/files/8754592/artefacts.zip)

Environment (please complete the following information):

OS: Linux
Toolchain: Zephyr SDK, 0.14.2
Commit SHA or Version used: https://github.com/zephyrproject-rtos/zephyr/commit/2dc866fa487ba524b426adc453d028bdd7aaa4e2

**Comment by mbolivar-nordic** [comment]
@gopiotr do you think you might be able to look into this?

**Comment by rljordan** [comment]
Anas is out of the office for a couple of weeks and cannot own this bug as Code Freeze approaches.  I am reassigning to @gopiotr  to drive this.  Please reassign if you cannot.

**Comment by hakehuang** [comment]
according to my test, there is no such issue, the only differences is on the pre-script, what's inside?

```
ubuntu@ubuntu-OptiPlex-7050:/home/shared/disk/zephyr_project/zephyr_test/zephyr$ scripts/twister -p mimxrt1060_evk --device-testing --west-runner=pyocd --west-flash="--board-id=022900000582988000000000000000000000000097969905,--erase" --device-serial /dev/ttyACM0  -o /home/ubuntu/report/rt1060 --inline-logs -v --ninja --jobs 4 --disable-warnings-as-errors --test samples/hello_world/sample.basic.helloworld --test samples/philosophers/sample.kernel.philosopher.stacks --test tests/kernel/common/kernel.common  -W
Renaming output directory to /home/shared/disk/zephyr_project/zephyr_test/zephyr/twister-out.3
INFO    - Zephyr version: v3.1.0-rc2-56-g138f399e6c6b
INFO    - JOBS: 4
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/shared/disk/zephyr_project/zephyr_test/zephyr/twister-out/testplan.json
INFO    - Using 'zephyr' toolchain.

Device testing on:

| Platform       | ID   | Serial device   |
|----------------|------|-----------------|
| mimxrt1060_evk |      | /dev/ttyACM0    |

INFO    - 3 test scenarios (3 configurations) selected, 0 configurations discarded due to filters.
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - 1/3 mimxrt1060_evk            samples/hello_world/sample.basic.helloworld        PASSED (device)
INFO    - 2/3 mimxrt1060_evk            samples/philosophers/sample.kernel.philosopher.stacks PASSED (device)
INFO    - 3/3 mimxrt1060_evk            tests/kernel/common/kernel.common                  PASSED (device)

INFO    - 3 of 3 test configurations passed (100.00%), 0 failed, 0 skipped with 0 warnings in 140.82 seconds
INFO    - In total 43 test cases were executed, 4 skipped on 1 out of total 456 platforms (0.22%)
INFO    - 0 test configurations executed on platforms, 3 test configurations were only built.

Hardware distribution summary:

| Board          | ID   |   Counter |
|----------------|------|-----------|
| mimxrt1060_evk |      |         3 |
INFO    - Saving reports...
INFO    - Writing JSON report /home/ubuntu/report/rt1060/twister.json
INFO    - Using 'zephyr' toolchain.
INFO    - Writing xunit report /home/ubuntu/report/rt1060/twister.xml...
INFO    - Writing xunit report /home/ubuntu/report/rt1060/twister_report.xml...
INFO    - Run completed

```

```
<testsuites><testsuite name="mimxrt1060_evk" timestamp="2022-05-25T10:57:23.958821" time="0.0" tests="47" failures="0" errors="0" skipped="4"><properties><property name="version" value="v3.1.0-rc2-56-g138f399e6c6b" /></properties><testcase classname="kernel.common" name="kernel.common.bootdelay" time="0.10" /><testcase classname="kernel.common" name="kernel.common.clock_cycle_64" time="0"><skipped type="skipped" message="ztest skip" /></testcase><testcase classname="kernel.common" name="kernel.common.sys_put_be64" time="0.10" /><testcase classname="kernel.common" name="kernel.common.nop" time="0"><skipped type="skipped" message="ztest skip" /></testcase><testcase classname="kernel.common" name="kernel.common.clock_uptime" time="0.12" /><testcase classname="kernel.common" name="kernel.common.ffs" time="0.10" /><testcase classname="kernel.common" name="kernel.common.atomic" time="0.10" /><testcase classname="kernel.common" name="kernel.common.timeout_order" time="0.13" /><testcase classname="kernel.common" name="kernel.common.threads_access_atomic" time="0.26" /><testcase classname="kernel.common" name="kernel.common.thread_context" time="0.31" /><testcase classname="kernel.common" name="kernel.common.sys_get_le48" time="0.10" /><testcase classname="kernel.common" name="kernel.common.printk" time="0.30" /><testcase classname="kernel.common" name="kernel.common.sys_get_le16" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_be48" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_get_le64" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_le16" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_get_be16" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_be24" time="0.10" /><testcase classname="kernel.common" name="kernel.common.multilib" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_le64" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_get_be24" time="0.10" /><testcase classname="kernel.common" name="kernel.common.byteorder_memcpy_swap" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_be16" time="0.10" /><testcase classname="kernel.common" name="kernel.common.bitarray_set_clear" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_le48" time="0.10" /><testcase classname="kernel.common" name="kernel.common.byteorder_mem_swap" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_be32" time="0.10" /><testcase classname="kernel.common" name="kernel.common.bitarray_alloc_free" time="0.44" /><testcase classname="kernel.common" name="kernel.common.errno" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_get_be32" time="0.10" /><testcase classname="kernel.common" name="kernel.common.bitarray_region_set_clear" time="0.80" /><testcase classname="kernel.common" name="kernel.common.bitarray_declare" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_le24" time="0.10" /><testcase classname="kernel.common" name="kernel.common.irq_offload" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_le32" time="0.10" /><testcase classname="kernel.common" name="kernel.common.ms_time_duration" time="0.18" /><testcase classname="kernel.common" name="kernel.common.nested_irq_offload" time="0"><skipped type="skipped" message="ztest skip" /></testcase><testcase classname="kernel.common" name="kernel.common.bounds_check_mitigation" time="0.10" /><testcase classname="kernel.common" name="kernel.common.bitfield" time="0"><skipped type="skipped" message="ztest skip" /></testcase><testcase classname="kernel.common" name="kernel.common.sys_get_be64" time="0.10" /><testcase classname="kernel.common" name="kernel.common.version" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_get_le24" time="0.10" /><testcase classname="kernel.common" name="kernel.common.clock_cycle_32" time="0.30" /><testcase classname="kernel.common" name="kernel.common.sys_get_be48" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_get_le32" time="0.10" /><testcase classname="sample.kernel" name="sample.kernel.philosopher.stacks" time="0.00" /><testcase classname="sample.basic" name="sample.basic.helloworld" time="0.00" /></testsuite></testsuites>

```

**Comment by hakehuang** [comment]
according to my test, there is no such issue, the only differences is on the pre-script, what's inside?

```
ubuntu@ubuntu-OptiPlex-7050:/home/shared/disk/zephyr_project/zephyr_test/zephyr$ scripts/twister -p mimxrt1060_evk --device-testing --west-runner=pyocd --west-flash="--board-id=022900000582988000000000000000000000000097969905,--erase" --device-serial /dev/ttyACM0  -o /home/ubuntu/report/rt1060 --inline-logs -v --ninja --jobs 4 --disable-warnings-as-errors --test samples/hello_world/sample.basic.helloworld --test samples/philosophers/sample.kernel.philosopher.stacks --test tests/kernel/common/kernel.common  -W
Renaming output directory to /home/shared/disk/zephyr_project/zephyr_test/zephyr/twister-out.3
INFO    - Zephyr version: v3.1.0-rc2-56-g138f399e6c6b
INFO    - JOBS: 4
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/shared/disk/zephyr_project/zephyr_test/zephyr/twister-out/testplan.json
INFO    - Using 'zephyr' toolchain.

Device testing on:

| Platform       | ID   | Serial device   |
|----------------|------|-----------------|
| mimxrt1060_evk |      | /dev/ttyACM0    |

INFO    - 3 test scenarios (3 configurations) selected, 0 configurations discarded due to filters.
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - 1/3 mimxrt1060_evk            samples/hello_world/sample.basic.helloworld        PASSED (device)
INFO    - 2/3 mimxrt1060_evk            samples/philosophers/sample.kernel.philosopher.stacks PASSED (device)
INFO    - 3/3 mimxrt1060_evk            tests/kernel/common/kernel.common                  PASSED (device)

INFO    - 3 of 3 test configurations passed (100.00%), 0 failed, 0 skipped with 0 warnings in 140.82 seconds
INFO    - In total 43 test cases were executed, 4 skipped on 1 out of total 456 platforms (0.22%)
INFO    - 0 test configurations executed on platforms, 3 test configurations were only built.

Hardware distribution summary:

| Board          | ID   |   Counter |
|----------------|------|-----------|
| mimxrt1060_evk |      |         3 |
INFO    - Saving reports...
INFO    - Writing JSON report /home/ubuntu/report/rt1060/twister.json
INFO    - Using 'zephyr' toolchain.
INFO    - Writing xunit report /home/ubuntu/report/rt1060/twister.xml...
INFO    - Writing xunit report /home/ubuntu/report/rt1060/twister_report.xml...
INFO    - Run completed

```

```
<testsuites><testsuite name="mimxrt1060_evk" timestamp="2022-05-25T10:57:23.958821" time="0.0" tests="47" failures="0" errors="0" skipped="4"><properties><property name="version" value="v3.1.0-rc2-56-g138f399e6c6b" /></properties><testcase classname="kernel.common" name="kernel.common.bootdelay" time="0.10" /><testcase classname="kernel.common" name="kernel.common.clock_cycle_64" time="0"><skipped type="skipped" message="ztest skip" /></testcase><testcase classname="kernel.common" name="kernel.common.sys_put_be64" time="0.10" /><testcase classname="kernel.common" name="kernel.common.nop" time="0"><skipped type="skipped" message="ztest skip" /></testcase><testcase classname="kernel.common" name="kernel.common.clock_uptime" time="0.12" /><testcase classname="kernel.common" name="kernel.common.ffs" time="0.10" /><testcase classname="kernel.common" name="kernel.common.atomic" time="0.10" /><testcase classname="kernel.common" name="kernel.common.timeout_order" time="0.13" /><testcase classname="kernel.common" name="kernel.common.threads_access_atomic" time="0.26" /><testcase classname="kernel.common" name="kernel.common.thread_context" time="0.31" /><testcase classname="kernel.common" name="kernel.common.sys_get_le48" time="0.10" /><testcase classname="kernel.common" name="kernel.common.printk" time="0.30" /><testcase classname="kernel.common" name="kernel.common.sys_get_le16" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_be48" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_get_le64" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_le16" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_get_be16" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_be24" time="0.10" /><testcase classname="kernel.common" name="kernel.common.multilib" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_le64" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_get_be24" time="0.10" /><testcase classname="kernel.common" name="kernel.common.byteorder_memcpy_swap" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_be16" time="0.10" /><testcase classname="kernel.common" name="kernel.common.bitarray_set_clear" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_le48" time="0.10" /><testcase classname="kernel.common" name="kernel.common.byteorder_mem_swap" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_be32" time="0.10" /><testcase classname="kernel.common" name="kernel.common.bitarray_alloc_free" time="0.44" /><testcase classname="kernel.common" name="kernel.common.errno" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_get_be32" time="0.10" /><testcase classname="kernel.common" name="kernel.common.bitarray_region_set_clear" time="0.80" /><testcase classname="kernel.common" name="kernel.common.bitarray_declare" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_le24" time="0.10" /><testcase classname="kernel.common" name="kernel.common.irq_offload" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_put_le32" time="0.10" /><testcase classname="kernel.common" name="kernel.common.ms_time_duration" time="0.18" /><testcase classname="kernel.common" name="kernel.common.nested_irq_offload" time="0"><skipped type="skipped" message="ztest skip" /></testcase><testcase classname="kernel.common" name="kernel.common.bounds_check_mitigation" time="0.10" /><testcase classname="kernel.common" name="kernel.common.bitfield" time="0"><skipped type="skipped" message="ztest skip" /></testcase><testcase classname="kernel.common" name="kernel.common.sys_get_be64" time="0.10" /><testcase classname="kernel.common" name="kernel.common.version" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_get_le24" time="0.10" /><testcase classname="kernel.common" name="kernel.common.clock_cycle_32" time="0.30" /><testcase classname="kernel.common" name="kernel.common.sys_get_be48" time="0.10" /><testcase classname="kernel.common" name="kernel.common.sys_get_le32" time="0.10" /><testcase classname="sample.kernel" name="sample.kernel.philosopher.stacks" time="0.00" /><testcase classname="sample.basic" name="sample.basic.helloworld" time="0.00" /></testsuite></testsuites>

```

**Comment by hakehuang** [comment]
> Full logs and twister reports:
> [artefacts.zip](https://github.com/zephyrproject-rtos/zephyr/files/8754592/artefacts.zip)

this zip file is empty

**Comment by chen-png** [comment]
>1/3 hsdk                      samples/hello_world/sample.basic.helloworld        PASSED (**device 63.116s)**

the detailed handler time "device 63.116s" won't be showed up when it's normal, it is only showed up when this test failed due to timeout. 
so I guess **you didn't get expected "Hello World!" from console actually,** the harness.state is None, so the handler time is about the default timeout time, the test should be failed due to timeout, but **due to twister issue, if harness.state is None, it won't set the instance.status again, then it will still keep the previous "passed" status from build stage, this is the reason why you see "PASSED" in output, but actually it should be FAILED due to Timeout.**

@evgeniy-paltsev Could you help check the handler.log of test "helloworld" in your twister-out directory, did you really get "Hello World!" string from your console?

**found a method to reproduce this, just changed the "Hello World" string to an incorrect string** in the main.c file of test "helloworld", like "hello world", then run it again, it should be failed due to timeout, but the console is passed.

```
twister -p reel_board --device-testing --device-serial /dev/ttyACM0 -T samples/hello_world/ -v
1/1 reel_board                samples/hello_world/sample.basic.helloworld        PASSED (device 63.488s)
```

**Comment by chen-png** [comment]
this twister issue will be fixed by PR #45943

**Comment by gopiotr** [comment]
As @hakehuang mentioned in this PR: https://github.com/zephyrproject-rtos/zephyr/pull/45943, this issue has two source of problem. First one is wrong Twister test marking - due to the timeout it should be marked as fail/error - this will be fixed by this  PR: https://github.com/zephyrproject-rtos/zephyr/pull/45943. Second one is a problem with board or hardware configuration - from some reasons `hsdk` platform do not print output properly.
